# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/preferences/privacy.js
# Commit: 201f0d5fabd6
# Full Hash: 201f0d5fabd61c8eebe7e27e8597021d3618ddea
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2021-03-29 15:25:22
# Regressor Bug: 1695050
# File Overlap Count: 1
# Description:
#   Bug 1695050 - Part 2: Modify CookieJarSettings::Create() to be able to creat cookieJarSettings according to the browsing mode. r=dimi,preferences-reviewers
#   
#   This patch modifies the current CookieJarSettings::Create() function.
#   It removes the current function and adds two variants. One takes the
#   nsIPrincipal as input and another takes the enum value. The new
# ==============================================================================

diff -r 59cda3c92f48 -r 201f0d5fabd6 browser/components/preferences/privacy.js
--- a/browser/components/preferences/privacy.js	Mon Mar 29 11:01:50 2021 +0000
+++ b/browser/components/preferences/privacy.js	Mon Mar 29 11:01:51 2021 +0000
@@ -1213,7 +1213,7 @@
    * Selects the right items of the new Cookies & Site Data UI.
    */
   networkCookieBehaviorReadPrefs() {
-    let behavior = Services.cookies.cookieBehavior;
+    let behavior = Services.cookies.getCookieBehavior(false);
     let blockCookiesMenu = document.getElementById("blockCookiesMenu");
     let deleteOnCloseCheckbox = document.getElementById("deleteOnClose");
     let deleteOnCloseNote = document.getElementById("deleteOnCloseNote");
@@ -1691,7 +1691,8 @@
   readBlockCookies() {
     let bcControl = document.getElementById("blockCookiesMenu");
     bcControl.disabled =
-      Services.cookies.cookieBehavior == Ci.nsICookieService.BEHAVIOR_ACCEPT;
+      Services.cookies.getCookieBehavior(false) ==
+      Ci.nsICookieService.BEHAVIOR_ACCEPT;
   },
 
   /**
@@ -1711,7 +1712,7 @@
   },
 
   readBlockCookiesFrom() {
-    switch (Services.cookies.cookieBehavior) {
+    switch (Services.cookies.getCookieBehavior(false)) {
       case Ci.nsICookieService.BEHAVIOR_REJECT_FOREIGN:
         return "all-third-parties";
       case Ci.nsICookieService.BEHAVIOR_REJECT: